pkg/controller/jobframework/reconciler.go

```diff
@@ -574,6 +595,44 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 	return ctrl.Result{}, nil
 }
 
+func shouldHandleDeletionOfDeactivatedWorkload(afterDeactivatedByKueue *time.Duration, job GenericJob, wl *kueue.Workload) bool {
+	return afterDeactivatedByKueue != nil &&
+		job.Object().GetDeletionTimestamp() == nil &&
+		!workload.IsActive(wl) &&
+		!workload.IsFinished(wl) &&
```

Comment:
When these lines are removed, unit tests still pass. Please add coverage that will catch if any of the 3 lines are removed

Reply:
Thank you for recheck!